nucypher/utilities/porter/control/specifications/fields/treasuremapid.py

```diff
@@ -19,11 +19,14 @@
 
 from nucypher.control.specifications.exceptions import InvalidInputData
 from nucypher.control.specifications.fields.base import BaseField
+from nucypher.crypto.constants import HRAC_LENGTH
 from nucypher.policy.collections import TreasureMap
 
 
 class TreasureMapID(BaseField, fields.String):
 
     def _validate(self, value):
-        if len(bytes.fromhex(value)) != TreasureMap.ID_LENGTH:
+        treasure_map_id = bytes.fromhex(value)
+        # FIXME federated has map id length 32 bytes but decentralized has length 16 bytes ... huh?
```

Comment:
Wait, aren't they different things? `TreasureMap` creates an ID based on HRAC (`self._id = keccak_digest(bytes(self._verifying_key) + bytes(self._hrac)).hex()`)

Reply:
Disregard that, apparently in federated mode it's just `hrac.hex()`. Not a great thing to have, should really be made less error-prone in some way. A `MapID` class, perhaps?

Reply:
`MapID` could be good but that affects what `TreasureMap.public_id()` should return - trying to keep my surface area of changes limited.

Reply:
```suggestion
        # FIXME federated has map id length 32 bytes but decentralized has length 16 bytes ... huh? - #2725
```
Opened issue #2725